x-pack/filebeat/module/crowdstrike/falcon_endpoint/config/pipeline.js

```diff
@@ -0,0 +1,126 @@
+
+var crowdstrikeFalcon = (function() {
+    var processor = require("processor");
+
+    var convertUnderscore = function(text) {
+        return text.split(/(?=[A-Z])/).join('_').toLowerCase(); 
+    };
+
+    var decodeJson = new processor.DecodeJSONFields({
+        fields: ["message"],
+        target: "crowdstrike",
+        process_array: true,
+        max_depth: 8
+    });
+
+    // Don't process non-endpoint related events handled by the endpoint fileset
+    // This avoids generating duplicate events since both event types come from 
+    // the same log file. Abort early to minimize CPU time.
+    var cancelNonEndpointEvents = function(evt) {
+        var eventType = evt.Get("crowdstrike.metadata.eventType")
+        switch (eventType) {
+            case "UserActivityAuditEvent":
+            case "AuthActivityAuditEvent":
+            case "RemoteResponseSessionStartEvent":
+            case "RemoteResponseSessionEndEvent":
+                evt.Cancel()
+                break;
+        }
+    };
+
+    var dropFields = function(evt) {
+        evt.Delete("message");
+        evt.Delete("host.name");
+    };
+  
+    var setFields = function (evt) {
+        evt.Put("agent.name", "falcon");
+    };
+
+    var convertFields = new processor.Convert({
+        fields: [
+            // DetectionSummaryEvent
+            { from: "crowdstrike.event.LocalIP", to: "source.ip", type: "ip" },
+            { from: "crowdstrike.event.ProcessId", to: "process.pid" },
+        ],
+        mode: "copy",
+        ignore_missing: true,
+        ignore_failure: true
+    });
+
+    var parseTimestamp = new processor.Timestamp({
+        field: "crowdstrike.metadata.eventCreationTime",
+        target_field: "@timestamp",
+        timezone: "UTC",
+        layouts: ["UNIX_MS"],
+        ignore_missing: false,
+    });
+
+    var processEvent = function(evt) {
+        var eventType = evt.Get("crowdstrike.metadata.eventType")
+
+        switch (eventType) {
+            case "DetectionSummaryEvent":
+                var tactic = evt.Get("crowdstrike.event.Tactic").toLowerCase()
+                var technique = evt.Get("crowdstrike.event.Technique").toLowerCase()
+                evt.Put("threat.technique.name", technique) 
+                evt.Put("threat.tactic.name", tactic)
+
+                evt.Put("event.action", evt.Get("crowdstrike.event.PatternDispositionDescription"))
+                evt.Put("event.kind", "alert")
+                evt.Put("event.type", ["info"])
+                evt.Put("event.category", ["malware"])
+                evt.Put("event.url", evt.Get("crowdstrike.event.FalconHostLink"))
+
+                evt.Put("event.severity", evt.Get("crowdstrike.event.Severity"))
+                evt.Put("message", evt.Get("crowdstrike.event.DetectDescription"))
+                evt.Put("process.name", evt.Get("crowdstrike.event.FileName"))
+                evt.Put("process.executable", evt.Get("crowdstrike.event.CommandLine"))
```

Comment:
Do you know if this value can contain arguments? The sample event doesn't have any args, but the name suggests that it may. I would suggest using `process.command_line` for this value. Then derive `process.executable` from the command_line by taking the first argument (like split(" ")[0]).